RFC-007-SWAP-Ethereum-Basic.md

```diff
@@ -0,0 +1,291 @@
+- How to construct a Hash Time Lock Contract (HTLC) to lock the Ether asset on the Ethereum blockchain.
+- How to deploy, redeem and refund the HTLC during the execution phase of [RFC003](./RFC-003-SWAP-basic.md).
+
+## The Ethereum Identity
+
+The Identity to be used on Ethereum is the Ethereum *address* as defined in section 284 of the [Ethereum Yellowpaper](https://ethereum.github.io/yellowpaper/paper.pdf).
+It is defined as the right most 160-bits (20 bytes) of the Keccak-256 hash of the corresponding ECDSA public key.
+
+In the JSON encoding, an ethereum address MUST be encoded as this 20 byte hex string prefixed by `0x` (as is standard in the Ethereum ecosystem).
+Furthermore, implementations MUST also accept [EIP50](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-55.md) mixed case addresses and MAY verify the checksum.
+
+This RFC extends the [registry](./registry.md) with the following entry in the identity table:
+
+| Ledger   | Identity Name | JSON Encoding         | Description         |
+|:---------|:--------------|:----------------------|---------------------|
+| Ethereum | `address`     | `0x` prefixed address | An Ethereum Address |
+
+Note that since *contract* addresses and *user* addresses are indistinguishable, technically the address used in a SWAP could be that of a contract rather than a user.
+RFCs that use the Ethereum Identity defined here should explain the impact (if any) this has on the protocol.
+There is nothing inherently wrong from the perspective of this RFC about using a contract address as an identity.
+
+## Hash Time Lock Contract
+
+### Hash Functions
+
+This RFC specifies SHA-256 is the only value the `hash_function` header may take if Ethereum is used as a ledger.
+This may be expanded in subsequent RFCs.
+
+### Parameters
+
+The parameters for the Ether HTLC follow [RFC003](./RFC-003-SWAP-basic.md) and are described concretely in the following table:
+
+| Variable        | Description                                                                 |
+|:----------------|:----------------------------------------------------------------------------|
+| asset           | The quantity of wei                                                         |
+| secret_hash     | The hash of the `secret` (32 bytes for SHA-256)                             |
+| redeem_identity | The `address` of the redeeming party                                        |
+| refund_identity | The `address` of the refunding party                                        |
+| expiry          | The absolute UNIX timestamp in seconds after which the HTLC can be refunded |
+
+### Contract
+
+This RFC defines a single disposable contract as the HTLC.
+It uses `selfdestruct` to release the funds to the intended party.
+
+This approach was chosen over a stateful contract written in solidity so the contracts can be precisely defined, verified and updated.
```

Comment:
```suggestion
This approach was chosen over a stateful contract written in Solidity to avoid having to deal with topics like upgradability or run into the risk of endangering the funds of multiple users due to a bug in the contract.
```

Reply:
How does not using a stateful contract not endanger the funds of multiple users due to a bug in the contract though?